packages/edit-site/src/components/pattern-actions/index.js

```diff
@@ -0,0 +1,297 @@
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	__experimentalConfirmDialog as ConfirmDialog,
+} from '@wordpress/components';
+import { moreVertical } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { decodeEntities } from '@wordpress/html-entities';
+import { store as reusableBlocksStore } from '@wordpress/reusable-blocks';
+import { downloadBlob } from '@wordpress/blob';
+import { getQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import isTemplateRemovable from '../../utils/is-template-removable';
+import isTemplateRevertable from '../../utils/is-template-revertable';
+import RenameMenuItem from './rename-menu-item';
+import DuplicateMenuItem from './duplicate-menu-item';
+import {
+	TEMPLATE_POST_TYPE,
+	TEMPLATE_PART_POST_TYPE,
+	POST_TYPE_LABELS,
+	PATTERN_TYPES,
+} from '../../utils/constants';
+
+export default function PatternActions( {
+	item,
+	className,
+	toggleProps = {},
+	popoverProps = {},
+	onRemove,
+	icon = moreVertical,
+} ) {
+	const record = useSelect(
+		( select ) => {
+			if ( item?.type && item?.id ) {
+				return select( coreStore ).getEntityRecord(
+					'postType',
+					item?.type,
+					item?.id
+				);
+			}
+			return null;
+		},
+		[ item?.type, item?.id ]
+	);
+	const isBlockBasedTheme = useSelect( ( select ) => {
+		return select( coreStore ).getCurrentTheme()?.is_block_theme;
+	}, [] );
+	const { removeTemplate, revertTemplate } = useDispatch( editSiteStore );
+	const { saveEditedEntityRecord } = useDispatch( coreStore );
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( noticesStore );
+	const { __experimentalDeleteReusableBlock } =
+		useDispatch( reusableBlocksStore );
+
+	// Don't show pattern actions for non-block based or hybrid themes.
+	if ( ! isBlockBasedTheme ) {
+		return null;
+	}
+
+	// Only custom patterns or custom template parts can be renamed or deleted.
+	const isUserPattern = record?.type === PATTERN_TYPES.user;
+	const isNonUserPattern = item?.type === PATTERN_TYPES.theme;
+	const isTemplate = record?.type === TEMPLATE_POST_TYPE;
+	const isTemplatePart = record?.type === TEMPLATE_PART_POST_TYPE;
+
+	// Type check to make sure we're dealing with a pattern or template.
+	if (
+		! isTemplatePart &&
+		! isTemplate &&
+		! isUserPattern &&
+		! isNonUserPattern
+	) {
+		return null;
+	}
+
+	const isRemovable = isTemplateRemovable( record );
+	const isRevertable = isTemplateRevertable( record );
+	// Only patterns and template parts can be duplicated for now.
+	const isDuplicable = isUserPattern || isNonUserPattern || isTemplatePart;
+
+	// If the pattern is not editable or duplicable, don't show the menu.
```

Comment:
```suggestion
	// If the pattern is not removable, revertable, or duplicable, don't show the menu.
```

The current comment doesn't quite seem to fit given `isEditable` is defined below and not used in the conditional.

Alternatively, is it worth doing away with the comment entirely as that code block is pretty self-explanatory now?

Reply:
I was using a editable to cover all the change of state conditions, but you're right: it's just confusing noise since the vars are self-explanatory. I think deleting is a good option. 

👍🏻